random_docs/Debugging monitoring stack on Openshift.md

```diff
@@ -270,7 +270,17 @@ And the output will be opened on the webwroser
 
 ## CPU Profile interpretation
 
-[ToDo]
+From the diagrams above, we can see Node Export spends most of its CPU time in collecting metrics by netClass and netDev collector.
+Especially in the netclass collector, the performance is dragged back by excessive file operations.
+
+The netclass collector walks in each directory in the sysfs mount point /sys/net/class/, read files and parse the content to numbers.
+As for network devices, reading some properties requires locking the RTNL lock in the Linux kernel, which competes for control over the network stack with kubelet that is busy setting up network for pods. 
+The busier the network stack is, the worse the performance for both node exporter and kubelet will become.
+
+The second CPU time consumer is netdev collector. In recent versions of Node Exporter, it is using the netlink interface to communicate with kernel for network device statistics. 
+Unlike ready the files in sysfs, communicating over netlink socket is processed asynchronically in the kernel, that does not interrupt kernel's current process to satisfy a syscall immediately. So we can see it  consumes less time than netclass collector.
+
+[An issue](https://github.com/prometheus/node_exporter/issues/2477) has been raised to the node exporter maintainer with [a proposed new collector](https://github.com/prometheus/node_exporter/pull/2492#issuecomment-1278188530) replacing the netclass and netdev collector using netlink interface for better performance.
```

Comment:
I propose to elaborate a little bit deeper this part. This document is not the only one that pointed of an excessive consumption by node-exporter. And there are new ideas coming, like the new collector implementation. Maybe we can link, to other possible features coming.